### ISSUE TITLE
Plugins config-file not added to .plist file

### ISSUE THREAD

**Issue by EinfachHans** [apache/cordova-ios#775]
# Bug Report

## Problem

I developed a Plugin, which has some `<config-file>` Entrys. When i build the App for ios, they are not appended to the .plist File.

## Information

For example i have the Following in the ios platform part of my plugin:
```
<config-file target="*-Info.plist" parent="UIBackgroundModes">
    <array>
        <string>bluetooth-central</string>
    </array>
</config-file>
```

After building, the .plist file does't include "bluetooth-central" in the "UIBackgroundModes". An other Plugin, that i use for Location, does the same for adding "location" and this works and appears.

Am i doing anything wrong?

But In the `ion.json` File, it is present:
```
{
    "xml": "<array><string>bluetooth-central</string></array>",
    "count": 1
}
```

I noticed that config files, defined in this Plugin (https://github.com/jeduan/cordova-plugin-facebook4), are not present in my plist either.

### Version information

I'm using `cordova-ios@5.1.1`

Maybe related to #581 , but it is closed (and merged), but still not works for me

**Comment by breautek** [comment]
Can you confirm you're using the latest version of the cli?

`cordova -v` should read..

```
9.0.0 (cordova-lib@9.0.1)
```

**Comment by HansKrywaa** [comment]
Yes, output is: `9.0.0 (cordova-lib@9.0.1)`

**Comment by HansKrywaa** [comment]
Can someone say me where this merging of the plugins.xml file happens in **cordova-ios**? Than i can try to fix it

**Comment by erisu** [comment]
The munging process is not in `cordova-ios`. It is in the core library `cordova-common`.

https://github.com/apache/cordova-common

**Comment by brodybits** [comment]
I just added the **help wanted** label, may be related to other issues linked in #793.

**Comment by dpa99c** [comment]
This is covered by https://github.com/apache/cordova-common/issues/144 and fixed by https://github.com/apache/cordova-common/pull/148

**Comment by HansKrywaa** [comment]
@dpa99c In which Version of cordova is this fixed?

**Comment by dpa99c** [comment]
@HansKrywaa [the fix](https://github.com/apache/cordova-common/commit/87167bd45670291b68e6e02852a6b4faf63ed247) is on the master branch of [cordova-common](https://github.com/apache/cordova-common) so is as yet unreleased (the latest npm release is [`cordova-common@4.0.1`](https://github.com/apache/cordova-common/tree/rel/4.0.1)).
The latest release `cordova-ios@6.1.0` [pins `cordova-common@^4.0.1](https://github.com/apache/cordova-ios/blob/rel/6.1.0/package.json#L43).

TL;DR: it will take a release of `cordova-common` then a release of `cordova-ios` which references it for this fix to become implicit in `cordova-ios`.

In the meantime, if you need to, you can install directly off [my fork of `cordova-ios`](https://github.com/dpa99c/cordova-ios) which contains this fix:

	cordova platform add https://github.com/dpa99c/cordova-ios

**Comment by HansKrywaa** [comment]
Ok understand 😃 No not that important, as i included all the stuff into my config.xml. Thanks for explaining!

**Comment by EinfachHans** [comment]
Closed as **cordova-common@4.0.2** is used since **cordova-ios@6.1.1** - thank you
